### ISSUE TITLE
data processing about "statement_label"

### ISSUE THREAD

**Issue by hellogirl007** [optimatch/optimatch#1]
Hi, thanks for sharing the wonderful work. How to generate the "statement_label" when using another data for the work? Why the length of "statement_label" is 465?

**Comment by wengcheng003** [comment]
> Hi, thanks for sharing the wonderful work. How to generate the "statement_label" when using another data for the work? Why the length of "statement_label" is 465?


the length of "statement_label" isn't it 155? The authors explain why in their paper:

We split the data into 80% for training, 10% for validation, and 10% for testing. For both our
approach and baselines, we consider n = 155 statements in each function and r = 20 tokens in
each statement as the descriptive statistics of the whole data set suggest that 95% of source code
functions have less than 155 statements and 95% of statements have less than 20 tokens. To initialize
our transformer encoders, we make use of the pre-trained model provided by Wang et al. [37]. This
model has undergone pre-training through various denoising objectives associated with programming
languages. Details of the hyperparameter settings for our method in both phases are provided in
Appendix A.4. In both training phases, we train our model through specific epochs and select the
model that demonstrates the highest F1 score for statement-level prediction in the validation set. The
experiments were conducted on a Linux machine with an AMD Ryzen 9 5950X processor, 64 GB of
RAM, and an NVIDIA RTX 3090 GPU. The potential limitations imposed by our experimental setup
are discussed in Appendix A.5.
